tests/test_infographic_api.py

```diff
@@ -0,0 +1,304 @@
+import pytest
+import six
+from six.moves import http_client
+from anyway import app as flask_app
+from jsonschema import validate
+
+
+@pytest.fixture
+def app():
+    return flask_app.test_client()
+
+
+if six.PY2:
+    def _text_data(rv): return rv.data
+else:
+    def _text_data(rv): return rv.data.decode("utf-8")
+
+
+class TestInfographicApi:
+    @pytest.fixture
+    def infographic_from_existing_new_flash(self, app):
+        EXITING_NEWS_FLASH_ID = 15305
+
+        rv = app.get(
+            f'/api/infographics-data?news_flash_id={EXITING_NEWS_FLASH_ID}')
+
+        return rv.get_json()
+
+    @pytest.fixture
+    def widgets(self, infographic_from_existing_new_flash):
+        return infographic_from_existing_new_flash['widgets']
+
+    def test_no_news_flash_id(self, app):
+        '''Should success and be empty when no flash id is sent'''
+        rv = app.get('/api/infographics-data')
+        assert rv.status_code == http_client.OK
+        assert not _text_data(rv)
```

Comment:
Those asserts will be might be optimized away, you should use `self.assert*` methods.

Reply:
No need, as we spoke about